inspec/a2-api-integration/controls/iam_v2.rb

```diff
@@ -427,63 +428,207 @@
+        resp = automate_api_request("/apis/iam/v2beta/users/#{CUSTOM_USER_ID}")
+        expect(resp.http_status).to eq 200
+        expect(resp.parsed_response_body[:user][:name]).to eq TEST_USER[:name]
+        expect(resp.parsed_response_body[:user][:id]).to eq TEST_USER[:id]
+      end
+
+      it "user gets a 404 when the user does not exist" do
+        resp = automate_api_request("/apis/iam/v2beta/users/some_wrong_id")
+        expect(resp.http_status).to eq 404
+      end
+    end
+
+    describe "PUT /apis/iam/v2beta/self/:id" do
+      it "user can update their own display name" do
+          updatedName =  "i updated my own name"
+          resp = automate_api_request("/apis/iam/v2beta/self/#{CUSTOM_USER_ID}",
+            http_method: 'PUT',
+            request_body: {
+              name: updatedName
+            }.to_json
+          )
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:user][:name]).to eq updatedName
+        end
+
+        it "user gets a 400 if their password is wrong" do
+          updatedName =  "i updated my own name"
+          resp = automate_api_request("/apis/iam/v2beta/self/#{CUSTOM_USER_ID}",
+            http_method: 'PUT',
+            request_body: {
+              name: updatedName,
+              password: "newpassword",
+              previous_password: "wrongagain"
+            }.to_json
+          )
+          expect(resp.http_status).to eq 400
+        end
+
+        # TODO (tc): right now, it returns 400 for a non-existent user because
+        # it checks for the password first, and if it doesn't match for a user
+        # that doesn't exist (always the case), it returns a 400.
+        it "returns 400 for a non-existent user" do
+          updatedName =  "i updated my own name"
+          resp = automate_api_request("/apis/iam/v2beta/self/some_wrong_id",
+            http_method: 'PUT',
+            request_body: {
+              name: updatedName,
+              password: "newpassword",
+              previous_password: "chefautomate"
+            }.to_json
+          )
+          expect(resp.http_status).to eq 400
+        end
+
+        it "user can update their own display name and password" do
+          updatedName =  "i updated my own name"
+          resp = automate_api_request("/apis/iam/v2beta/self/#{CUSTOM_USER_ID}",
+            http_method: 'PUT',
+            request_body: {
+              name: updatedName,
+              password: "newpassword",
+              previous_password: "chefautomate"
+            }.to_json
+          )
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:user][:name]).to eq updatedName
+        end
+      end
+
+      describe "DELETE /iam/v2beta/users/:id" do
+        it "deletes the user if it exists" do
+          resp = automate_api_request("/apis/iam/v2beta/users/#{CUSTOM_USER_ID}", http_method: 'DELETE')
+          expect(resp.http_status).to eq 200
```

Comment:
For testing `DELETE`, first verifying that the user exists, deleting, then checking that it's gone would be more robust, here. 

Recently we were getting `200s` even when the object wasn't being deleted.

Reply:
I'm asserting that it's a 200 on create in the before block but I can add that.

Reply:
Yeah it's that we were getting a `200` for a DELETE when the delete wasn't actually deleting-- that's why the `GET` check afterward helps also, since we're not using the db to insert. (I don't see the `GET` check yet)